Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM python:3
+ARG UNAME=wikibot
+ARG UID=1000
+ARG GID=1000
+RUN groupadd -g $GID -o $UNAME
+RUN useradd -m -u $UID -g $GID -o -s /bin/bash $UNAME
+USER $UNAME
```

Comment:
Why are we mounting anything in when running? You copy on build, and use --env-file for the env file

Doing uid stuff prevents us from building the image to the ghcr.io registry

Reply:
Also can we just mount source dir in for dev, so the docker doesn't write to disk?

Reply:
To follow up on an IRL conversation:

I still cannot find a way for one Dockerfile to directly reference another. You can use `FROM foo AS bar` to create names that `docker build --target bar` can reference in the same file.

If you're going to insist on using root in production, my recommendation is to adjust this code to default to root. Something like

```Dockerfile
ARG UNAME=0
ARG UID=0
ARG GID=0
RUN [ $GID == 0 || groupadd -g $GID -o $UNAME
RUN [ $UID == 0 || useradd -m -u $UID -g $GID -o -s /bin/bash $UNAME
USER $UNAME
```

This keeps DRY while making sure all the things are a consistent user (i fully expect pywikibot to behave Weird:tm: if the install and runtime users are different).